src/codegen/doc/mod.rs

```diff
@@ -0,0 +1,236 @@
+        |w| generate_doc(w, &env.library));
+}
+
+fn generate_doc(mut w: &mut Write, lib: &Library) -> Result<()> {
+    let namespace = lib.namespace(MAIN);
+
+    /*if let Some(ref namespace_doc) = namespace.doc {
+        for line in namespace_doc.split("\n") {
+            output.push_str(&format!("{}{}\n", FILE_COMMENT, line));
+        }
+    }
+    try!(create_sub_docs(&mut output, &namespace.constants, None));*/
+    for ty in namespace.types.iter().filter_map(|t| t.as_ref()) {
+        try!(handle_type(&mut w, &ty, None, lib));
+    }
+    Ok(())
+}
+
+fn handle_type(w: &mut Write, ty: &library::Type, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    match *ty {
+        LType::Alias(ref a) => {
+            try!(writeln!(w, "{}src/auto/{}.rs", FILE, a.name.to_lowercase()));
+            create_sub_doc(w, a, parent)
+        },
+        LType::Enumeration(ref e) => create_enum_doc(w, &e, parent),
+        LType::Function(ref f) => create_fn_doc(w, &f, parent),
+        LType::Interface(ref i) => create_trait_doc(w, &i, parent),
+        LType::Class(ref c) => create_class_doc(w, &c, parent, lib),
+        _ => Ok(()),
+    }
+}
+
+fn create_class_doc(w: &mut Write, class: &library::Class, parent: Option<Box<TypeStruct>>, lib: &Library) -> Result<()> {
+    let tabs : String = iter::repeat(" ").take(compute_ident(&parent) * TAB_SIZE).collect::<Vec<&str>>().concat();
+    let mut ty = class.convert();
+    ty.parent = parent;
+
+    try!(writeln!(w, "{}src/auto/{}.rs", FILE, class.name.to_lowercase()));
+    let found = if let Some(ref class_doc) = class.doc {
+        try!(writeln!(w, "{}{}\n{}",
+                      MOD_COMMENT,
+                      ty,
+                      format!("{}/// {}",
+                              tabs,
+                              class_doc.split("\n").collect::<Vec<&str>>().join(&format!("\n{}/// ", tabs)))));
+        true
+    } else {
+        false
+    };
+
+    if let Some(ref class_doc) = class.doc_deprecated {
+        if found == false {
+            try!(writeln!(w, "{}{}",
+                          MOD_COMMENT,
+                          ty));
+        }
+        try!(writeln!(w, "{}{}\n{}/// {}",
+                      tabs,
+                      DEPRECATED,
+                      tabs,
+                      class_doc.split("\n").collect::<Vec<&str>>().join(&format!("\n{}/// ", tabs))));
+    }
+    for i in 0..class.functions.len() {
+        try!(create_fn_doc(w, &class.functions[i], Some(Box::new(ty.clone()))));
+    }
+    for child_id in class.children.iter() {
+        if let &LType::Class(ref child) = lib.type_(*child_id) {
+            try!(create_class_doc(w, &child, None, lib));
+        }
+    }
+    Ok(())
+}
+
+fn create_trait_doc(w: &mut Write, trait_: &library::Interface, parent: Option<Box<TypeStruct>>) -> Result<()> {
+    let tabs : String = iter::repeat(" ").take(compute_ident(&parent) * TAB_SIZE).collect::<Vec<&str>>().concat();
```

Comment:
I'd say this is an improvement on what I pointed at because I was just talking about constants and this functions gives you the whole string at once.